x/staking/autocli.go

```diff
@@ -0,0 +1,140 @@
+package staking
+
+import (
+	"fmt"
+
+	autocliv1 "cosmossdk.io/api/cosmos/autocli/v1"
+	v1beta "cosmossdk.io/api/cosmos/staking/v1beta1"
+
+	"github.com/cosmos/cosmos-sdk/version"
+)
+
+func (am AppModule) AutoCLIOptions() *autocliv1.ModuleOptions {
+	return &autocliv1.ModuleOptions{
+		Query: &autocliv1.ServiceCommandDescriptor{
+			Service: v1beta.Query_ServiceDesc.ServiceName,
+			RpcCommandOptions: []*autocliv1.RpcCommandOptions{
+				{
+					RpcMethod: "Validators",
+					Short:     "Query for all validators",
+					Long:      "Query details about all validators on a network.",
+					Example:   fmt.Sprintf("$ %s query staking validators", version.AppName),
+				},
+				{
+					RpcMethod: "Validator",
+					Use:       "validator [validator-addr]",
```

Comment:
Getting `6:29PM ERR failure when running app err="invalid argument \"cosmosvaloper1hzktuc7un3uvrem2fdh2kw4vwzkxum6pv83h5c\" for \"--0\" flag: invalid bech32 account address: hrp does not match bech32 prefix: expected 'cosmos' got 'cosmosvaloper': internal logic error [cosmos/cosmos-sdk@v0.51.0/x/auth/codec/bech32_codec.go:37]"` when trying to get a validator.

Reply:
Fixed by https://github.com/cosmos/cosmos-sdk/issues/16767